src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/ModuleFreeCam.kt

```diff
@@ -89,18 +92,27 @@ object ModuleFreeCam : Module("FreeCam", Category.PLAYER) {
     val packetHandler = handler<PacketEvent> { event ->
         when (val packet = event.packet) {
             // For better FreeCam detecting AntiCheats, we need to prove to them that the player's moving
-            is PlayerMoveC2SPacket.PositionOnly, is PlayerMoveC2SPacket.LookOnly, is PlayerMoveC2SPacket.Both -> {
+            is PlayerMoveC2SPacket -> {
                 if (spoofMovement) {
-                    network.sendPacket(PlayerMoveC2SPacket(fakePlayer!!.isOnGround))
+                    if (packet.changePosition) {
```

Comment:
I would send no movement packets at all, because you're not moving. Just make sure changePosition and changeLook is false. Ok? AntiCheats detect if you send a movement packet for the same position you're standing on, which should never happen.

Reply:
Canceling changePosition and changeLook packets would send this on an advanced FreeCam detecting anticheat like BAC:

![Base Profile Screenshot 2021 05 02 - 15 25 59 80](https://user-images.githubusercontent.com/70612656/116813078-b8287e80-ab5a-11eb-9806-9fdb98fff168.png)

Reply:
@mems01 Is right. The `PlayerMoveC2SPacket` tells the server that the client has run a tick and not that the player has moved. Only it's variations send information about rotation and position.

Reply:
So? I guess I don't touch the code?